presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -689,7 +689,23 @@ folly::Future<std::unique_ptr<Result>> TaskManager::getResults(
       VELOX_USER_FAIL("Calling getResult() on a aborted task: {}", taskId);
     }
     if (prestoTask->error != nullptr) {
-      VELOX_USER_FAIL("Calling getResult() on a failed task: {}", taskId);
+      try {
```

Comment:
It is a good idea to provide more details here.
Will we see that extra message in query results (UI, Scuba)? Or only in the logs?

Do we need "std::rethrow_exception(prestoTask->error);" and catch?
Can we get the "what()" from the std::exception_ptr?
Also do we need a "catch (...)" when there is "const std::exception& e"?
Are there any other exceptions are flying around from our dependencies rather than std::exception?
"std::exception_ptr" hints that is is "std::exception", I assume.

Reply:
1. Yeah we will be able to see messages from scuba.
2. Yeah we need to throw and catch. There does not seem to be other ways.
3. std::exception_ptr can wrap around std::exceptions or any other class type.
4. I am not sure, the last catch is a safeguard in case there is some mysterious throw